include/kernel/thread.h

```diff
@@ -171,7 +173,25 @@ struct _thread_userspace_local_data {
 
 typedef struct k_thread_runtime_stats {
 #ifdef CONFIG_SCHED_THREAD_USAGE
+	/*
+	 * In the context of thread statistics, [execution_cycles] is the same
+	 * as the total # of non-idle cycles. In the context of CPU statistics,
+	 * it refers to the sum of non-idle + idle cycles.
+	 */
+
 	uint64_t execution_cycles;
+	uint64_t current_cycles;      /* current # of non-idle cycles */
+	uint64_t peak_cycles;         /* peak # of non-idle cycles */
+	uint64_t total_cycles;        /* total # of non-idle cycles */
+	uint64_t average_cycles;      /* average # of non-idle cycles */
```

Comment:
Also: "average_cycles" needs docs.  It looks like it's not an average over time but the "average length consumed time quantum" (i.e. on average how long does it run before it gets scheduled away).  That doesn't correspond to a number I'm familiar with from other OSes.  Might be useful, but really needs clarity.

Similarly you might consider renaming "peak" (which again tends to connote some kind of "rate" or time domain thing) to "longest" (which is what it actually stores: the longest time this thread got to run).

Finally: note that z_sched_usage_stop()/start() are called in a pair for every ISR.  So... those current/average/peak numbers are going to be polluted by whatever the interrupt behavior of the hardware devices is.  They don't really reflect thread behavior alone.  Is that intentional?

Reply:
Ah, OK.  Ignore the note about ISR.  I missed that you'd added an en/disable API, and that's where the statistics boundaries are placed.  That seems fine to me.

Reply:
Docs needed on average, and renaming peak to longest -- makes sense to me. It shall be so.